### PR TITLE
Added Channel Interface for Telemetry

### LINKED ISSUE #2585: Create Telemetry/Channel Interface
| | |
|:---|:---|
|**_F´ Version_**| |
|**_Affected Component_**|  |
---
## Feature Description

Make an interface out of the standard Channel ports. See #2255 for an example interface change set.

The following three ports should be put into an interface at `Fw/Interfaces/ChannelInterface.fppi` such that component implementers can quickly use them.

https://github.com/nasa/fprime/blob/a73bff28617b96736c0e3e222e4c6e1dea1c6f5e/Ref/SignalGen/SignalGen.fpp#L56-L57

### PR DESCRIPTION
**Pull request by menaman123**
| | |
|:---|:---|
|**#2585**|  |
|**No**|  |
|**No**|  |

---
## Change Description

Added ChannelInterface.fppi and included the telemetry within it.

## Rationale

Part of #2578.

## Future Work

Work on other interfaces

### CODE REVIEW

**Comment by thomas-bc** [comment]
Some tests are failing on `devel` branch: https://github.com/nasa/fprime/actions/runs/9357605666/job/25757702996
An oversight on our end.

Would you please add `CommandInterface.fppi` `ChannelInterface.fppi` and `EventInterface.fppi` to 
- https://github.com/nasa/fprime/blob/7d10786a10339d22bacb315f733296fb4b3465c3/cmake/test/src/test_unittests.py#L133
- https://github.com/nasa/fprime/blob/7d10786a10339d22bacb315f733296fb4b3465c3/cmake/test/src/test_ref_shared.py#L112

**Review by thomas-bc — COMMENTED** [review]
Some tests are failing on `devel` branch: https://github.com/nasa/fprime/actions/runs/9357605666/job/25757702996
An oversight on our end.

Would you please add `CommandInterface.fppi` `ChannelInterface.fppi` and `EventInterface.fppi` to 
- https://github.com/nasa/fprime/blob/7d10786a10339d22bacb315f733296fb4b3465c3/cmake/test/src/test_unittests.py#L133
- https://github.com/nasa/fprime/blob/7d10786a10339d22bacb315f733296fb4b3465c3/cmake/test/src/test_ref_shared.py#L112

**Comment by thomas-bc** [comment]
@LeStarch we probably should extend triggers on https://github.com/nasa/fprime/blob/devel/.github/workflows/cmake-test.yml. The tests are pretty quick so I would be in favor of making it run on every PR

File: Drv/LinuxSpiDriver/LinuxSpiDriver.fpp [hunk 1]
```diff
@@ -6,8 +6,8 @@ module Drv {
     # General ports
     # ----------------------------------------------------------------------
 
-    sync input port SpiReadWrite: Drv.SpiReadWrite
-
+    sync input port SpiReadWrite: Drv.SpiReadWrite 
+
```
Comment (menaman123) [inline Drv/LinuxSpiDriver/LinuxSpiDriver.fpp]: This is from a different commit that i did not want to be added to this PR

File: cmake/test/src/test_ref_shared.py [hunk 2]
```diff
@@ -109,7 +109,15 @@ def test_ref_module_info(REF_BUILD):
...
     ], "Did not find expected sources"
-    expected_ac = ["SignalGen.fpp", "Commands.fppi", "Events.fppi", "Telemetry.fppi"]
+    expected_ac = [
+        "SignalGen.fpp", 
+        "Commands.fppi", 
+        "Events.fppi", 
+        "Telemetry.fppi",
+        "CommandInterface.fppi", 
+        "ChannelInterface.fppi", 
+        "EventsInterface.fppi",
```
Comment (thomas-bc) [inline cmake/test/src/test_ref_shared.py:119]: We should probably choose between singular or plural for the 3 interfaces here and adapt filenames accordingly

**Comment by menaman123** [comment]
@thomas-bc do you happen to know why the format keeps failing?

**Comment by LeStarch** [comment]
@menaman123 format is failing because you have spaces ` ` after the commas in that modified python file.

@thomas-bc yes, we can run cmake on every commit.  It is faster now.

**Comment by menaman123** [comment]
@LeStarch I think i fixed the spaces. I hope it doesnt fail again

File: cmake/test/src/test_ref_shared.py (hunk 2 above)
Comment (LeStarch) [reply cmake/test/src/test_ref_shared.py:119]: Yes, we probably should rename `Events` interface to `Event` interface...but that would be another PR, no?

**Review by LeStarch — APPROVED** [review]

**Comment by LeStarch** [comment]
I went ahead and touched up the formatting and reverted the file you did not want.  Hopefully it will build cleanly now.

**Pull request merged**